packages/salesforcedx-vscode-core/test/vscode-integration/commands/templates/forceFunctionCreate.test.ts

```diff
@@ -10,103 +10,141 @@ import * as cp from 'child_process';
 import * as path from 'path';
 import { assert, createSandbox, SinonSandbox, SinonStub } from 'sinon';
 import * as vscode from 'vscode';
-import { ForceFunctionCreateExecutor } from '../../../../src/commands/templates/forceFunctionCreate';
-import { FUNCTION_TYPE_JAVA, FUNCTION_TYPE_JS } from '../../../../src/commands/templates/metadataTypeConstants';
+import {
+  forceFunctionCreate,
+  ForceFunctionCreateExecutor,
+  FunctionInfoGatherer
+} from '../../../../src/commands/templates/forceFunctionCreate';
+import {
+  FUNCTION_TYPE_JAVA,
+  FUNCTION_TYPE_JS
+} from '../../../../src/commands/templates/metadataTypeConstants';
+import { SfdxWorkspaceChecker } from '../../../../src/commands/util/preconditionCheckers';
 import { nls } from '../../../../src/messages';
 import { notificationService } from '../../../../src/notifications';
 import { telemetryService } from '../../../../src/telemetry';
 import * as rootWorkspace from '../../../../src/util';
 
+import * as library from '@heroku/functions-core';
+import sinon = require('sinon');
 // tslint:disable:no-unused-expression
 describe('Force Function Create', () => {
-  describe('Creation', () => {
-    it('Should build apex function create command for javascript', async () => {
-      const funcCreate = new ForceFunctionCreateExecutor();
-      const fileName = 'myFunc1';
-      const funcCreateCmd = funcCreate.build({
-        fileName,
-        language: 'javascript',
-        outputdir: ''
-      });
-
-      expect(funcCreateCmd.toCommand()).to.equal(
-        'sfdx generate:function --name myFunc1 --language javascript'
-      );
-      expect(funcCreateCmd.description).to.equal(
-        nls.localize('force_function_create_text')
-      );
-      expect(funcCreate.getFileExtension()).to.equal('.js');
+  let execStub: SinonStub;
+  const sandbox = createSandbox();
+  let notificationServiceStub: SinonStub;
+  let telemetryServiceStub: SinonStub;
+  let withProgressStub: SinonStub;
+  let openTextDocumentStub: SinonStub;
+  let showTextDocumentStub: SinonStub;
+  let gatherStub: SinonStub;
+  const functionInfoJS = {
+    type: 'CONTINUE' as 'CONTINUE',
```

Comment:
This hurts my eyes a little bit, but I get a typescript error when I don't include `as CONTINUE`